scripts/external_api_test.py

```diff
@@ -66,33 +73,36 @@ def main():
                     urls[url].add(value['legacy'])
 
                     # v3 urls
-                    url = (
-                        '%s/v3/links/fetch/%s/%s' %
-                        (onyx, locale, channel)
-                    )
-                    if url not in urls:
-                        urls[url] = set()
-                    urls[url].add(value['ag'])
+                    for release in release_names:
+                        url = (
+                            '%s/v3/links/fetch/%s/%s' %
+                            (onyx, locale, release)
+                        )
+                        if url not in urls:
+                            urls[url] = set()
+                        urls[url].add(value['ag'])
 
             # request urls
             results = grequests.map(
                 grequests.get(url, allow_redirects=False)
-                for url in urls.keys()
-            )
+                for url in urls.keys())
 
             # validate results
             for r in results:
-                if r.status_code != 303:
-                    print('ERROR: %s %s' % (r.url, r.status_code))
-                elif r.headers['location'] not in urls[r.url]:
-                    print(
-                        'ERROR: %s %s != %s' %
-                        (r.url, r.headers['location'], list(urls[r.url]))
-                    )
-                elif options.verbose:
-                    print('SUCCESS: %s %s' % (r.url, r.status_code))
-        except Exception as e:
-            print e
+                if r:
```

Comment:
Does grequests return a None somehow if failed? Just checked the doc, it does not specify.